widget/embedded/src/store/slices/data.ts

```diff
@@ -96,12 +99,38 @@ export const createDataSlice: StateCreator<
       type: options.type,
     });
 
+    const includedTokens = new Set();
+    const excludedTokens = new Set();
+
+    if (supportedTokensConfig) {
+      if (Array.isArray(supportedTokensConfig)) {
+        supportedTokensConfig.forEach((token) =>
+          includedTokens.add(createTokenHash(token))
+        );
+      } else if (!Array.isArray(supportedTokensConfig)) {
+        Object.values(supportedTokensConfig).forEach((value) => {
+          value.tokens.forEach((token) => {
+            if (token.blockchain === options.blockchain) {
+              if (value.isExcluded) {
+                excludedTokens.add(createTokenHash(token));
+              } else {
+                includedTokens.add(createTokenHash(token));
+              }
+            }
+          });
+        });
+      }
+    }
+
     const list = tokensFromState
       .filter((token) => {
-        if (
-          supportedTokensConfig &&
-          isTokenExcludedInConfig(token, supportedTokensConfig)
-        ) {
+        const tokenExcluded =
```

Comment:
In the configuration, we selected only two blockchains and limited each to a few tokens.
When we select all blockchains in the widget, it will show all tokens instead of showing the selected tokens.